### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by ocket8888**
#### What does this PR do?

Various documentation changes.

- Tables in reST source will not take up more than 215 characters per line
- Space indentation will be replaced with tabs
- Bold-faced terms and phrases followed by block-quoted text will be replaced with definition lists or sections as appropriate
- Instructions on how to accomplish things with the Traffic Ops API will be replaced with their Traffic Portal equivalents
- Undocumented API endpoints/parameters will be documented
- Grammatical, spelling and spacing issues will be rectified
- Sections requiring further explanation will be expanded upon

One thing I'm *not* planning on is updating any documentation related to the Traffic Ops UI. I'll still adjust the formatting, though.

#### Which TC components are affected by this PR?

- [x] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [ ] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] Other _________

#### What is the best way to verify this PR?

Build the documentation by running `make` in the `docs/` directory.
If you're missing dependencies, first run `sudo -H pip install -r docs/source/requirements.txt`

#### Check all that apply

- [ ] This PR includes tests
- [x] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [x] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2443/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2444/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2445/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2446/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2447/
Test FAILed.

File: docs/source/_templates/layout.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{% extends "!layout.html" %}
```
Comment (dangogh) [inline docs/source/_templates/layout.html]: missing the Apache header here

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2448/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2450/
Test PASSed.

**Pull request merged**